protocol/0082-RFPR-on_chain_referral_program.md

```diff
@@ -0,0 +1,235 @@
+| ------------------ | ---------------- | --------------------------------------------------------- | ---------------------------------------------------- |
+| `STATUS_PROPOSED`  | No               | Governance proposal valid and accepted                    | Governance proposal voting period ends               |
+| `STATUS_REJECTED`  | No               | Governance vote fails                                     | na                                                   |
+| `STATUS_PENDING`   | No               | Governance vote passes                                    | Current epoch greater than or equal to `start_epoch` |
+| `STATUS_STOPPED`   | No               | There is already an existing program with `STATUS_ACTIVE` | na                                                   |
+| `STATUS_ACTIVE`    | Yes              | Previously `STATUS_PENDING`                               | Current epoch less than or equal to `end_epoch`      |
+| `STATUS_CLOSED`    | No               | Previously `STATUS_ACTIVE`                                | na                                                   |              
+
+## Team Mechanics
+A team is comprised of a referrer and all their referees. There can only ever be one referrer per team but the number of referees is unlimited.
+
+### Creating a team
+
+To create a team and generate a referral code, a party must fulfil the following criteria:
+
+- party must not currently be a **referrer**
+- party must not currently be a **referee**
+- party must be staking at least `referralProgram.minStakedVegaTokens` tokens.
+
+This staking requirement is constant. If a referrer un-stakes enough tokens to fall below the requirement, they and their referees will no long be eligible for program benefits. If the referrer re-stakes enough tokens to fulfil the staking requirement, the team will become eligible for referral program benefits.
+
+A separate URL hosted on IPFS should allow a party to connect their Vega wallet and fill in the following optional team information:
+- name: an optional team name to be added to the referral banner.
+- url: an optional link to a team forum, discord, etc.
+- image: optional image data to be used when creating the referral banner.
+
+The party can then generate a team id by approving a transaction generated by the dApp. This team id should be associated with referrers public key by the network.
+
+As well as generating a referral code the dApp should then generate a banner with the referrers optional fields which they can use to invite referees.
+
+### Joining a team
+
+To become a referee, a referee must create a key (programmatically or through a dApp) and then send a transaction to associate their key with an existing team.
+
+message JoinTeam{
+    id: "abcd3fgh1jklmn0pqf5tuvwzyz"
+}
+
+A user should also be able to do this process by approving a transaction from a dApp.
+
+### Team epoch and running volumes
```

Comment:
More `quantum` scaling ...

Reply:
Ah, I see the `quantum` scaling now; so maybe just a clarification near the top of the spec that when we talk of volume we mean it's scaled by quantum already.